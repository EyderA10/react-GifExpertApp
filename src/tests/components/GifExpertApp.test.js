import React from 'react';
const { shallow } = require("enzyme")
const { default: GifExpertApp } = require("../../GifExpertApp")

describe('Pruebas con el componente GifExpertApp', () => {
    test('debe mostrarse correctamente', () => {
        const wrapper = shallow(<GifExpertApp />);

        expect(wrapper).toMatchSnapshot();
    })
    
    test('debe de mostrar una lista de categorias', () => {
        const categories = ['Dragon ball', 'game of thrones'];
        const wrapper = shallow(<GifExpertApp  defaultCategories={categories}/>);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length)
    })
    

})
