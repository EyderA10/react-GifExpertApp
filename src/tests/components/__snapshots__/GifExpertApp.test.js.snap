// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas con el componente GifExpertApp debe de mostrar una lista de categorias 1`] = `
<div>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Dragon ball"
      key="Dragon ball"
    />
    <GifGrid
      category="game of thrones"
      key="game of thrones"
    />
  </ol>
</div>
`;

exports[`Pruebas con el componente GifExpertApp debe mostrarse correctamente 1`] = `
<div>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</div>
`;
